BUG: Cannot set host relative to site root

Works:
```
setBuildURLConfig({
  host: 'https://mydomain.com',
namespace: 'api/v1'
});
```

Does not work:
```
setBuildURLConfig({
  host: '/',
  namespace: 'api/v1'
});
```

in route `mydomain.com/admin/users`:
`findRecord('users', 123) => mydomain.com/admin/users/api/v1/users/123` 

Expected 
`findRecord('users', 123) => mydomain.com/api/v1/users/123` 


Note that this only appears to work when using the index route


### Tests

```
setBuildURLConfig({
  namespace: 'api/v1'
});
```
Fails with `router.js:1142 Error while processing route: auth.tenants.tenant.index Cannot read properties of undefined (reading 'endsWith') `
in buildBaseURL
`(true && !(host === '/' || !host.endsWith('/')) && (0, _debug.assert)(`buildBaseURL: host must NOT end with '/', received '${host}'`, host === '/' || !host.endsWith('/')));`
    
------------------------------------------------
```
setBuildURLConfig({
  host: '',
  namespace: 'api/v1'
});
```
generates relative url `/api/v1/...`
Works correctly as expected but not intuitive for developer

------------------------------------------------
```
setBuildURLConfig({
  host: '/',
  namespace: 'api/v1'
});
```
generates relative url `api/v1/...`

--- a/packages/request-utils/src/index.ts
+++ b/packages/request-utils/src/index.ts
@@ -57,7 +57,7 @@ export interface BuildURLConfig {
   namespace: string | null;
 }
 
-let CONFIG: BuildURLConfig = {
+const CONFIG: BuildURLConfig = {
   host: '',
   namespace: '',
 };
@@ -69,13 +69,30 @@ let CONFIG: BuildURLConfig = {
  * These values may still be overridden by passing
  * them to buildBaseURL directly.
  *
- * This method may be called as many times as needed
+ * This method may be called as many times as needed.
+ * host values of `''` or `'/'` are equivalent.
+ *
+ * Except for the value of `/` as host, host should not
+ * end with `/`.
+ *
+ * namespace should not start or end with a `/`.
  *
  * ```ts
  * type BuildURLConfig = {
  *   host: string;
  *   namespace: string'
  * }
+ * ```
+ *
+ * Example:
+ *
+ * ```ts
+ * import { setBuildURLConfig } from '@ember-data/request-utils';
+ *
+ * setBuildURLConfig({
+ *   host: 'https://api.example.com',
+ *   namespace: 'api/v1'
+ * });
  * ```
  *
  * @method setBuildURLConfig
@@ -86,7 +103,15 @@ let CONFIG: BuildURLConfig = {
  * @return void
  */
 export function setBuildURLConfig(config: BuildURLConfig) {
-  CONFIG = config;
+  assert(`setBuildURLConfig: You must pass a config object`, config);
+  assert(`setBuildURLConfig: You must pass a config object with a 'host' or 'namespace' property`, 'host' in config || 'namespace' in config);
+
+  CONFIG.host = config.host || '';
+  CONFIG.namespace = config.namespace || '';
+
+  assert(`buildBaseURL: host must NOT end with '/', received '${CONFIG.host}'`, CONFIG.host === '/' || !CONFIG.host.endsWith('/'));
+  assert(`buildBaseURL: namespace must NOT start with '/', received '${CONFIG.namespace}'`, !CONFIG.namespace.startsWith('/'));
+  assert(`buildBaseURL: namespace must NOT end with '/', received '${CONFIG.namespace}'`, !CONFIG.namespace.endsWith('/'));
 }
 
 export interface FindRecordUrlOptions {
@@ -317,8 +342,9 @@ export function buildBaseURL(urlOptions: UrlOptions): string {
   assert(`buildBaseURL: idPath must NOT start with '/', received '${idPath}'`, !idPath.startsWith('/'));
   assert(`buildBaseURL: idPath must NOT end with '/', received '${idPath}'`, !idPath.endsWith('/'));
 
-  const url = [host === '/' ? '' : host, namespace, resourcePath, idPath, fieldPath].filter(Boolean).join('/');
-  return host ? url : `/${url}`;
+  const hasHost = host !== '' && host !== '/';
+  const url = [hasHost ? host : '', namespace, resourcePath, idPath, fieldPath].filter(Boolean).join('/');
+  return hasHost ? url : `/${url}`;
 }
 
 const DEFAULT_QUERY_PARAMS_SERIALIZATION_OPTIONS: QueryParamsSerializationOptions = {

--- a/packages/request-utils/src/index.ts
+++ b/packages/request-utils/src/index.ts
@@ -104,14 +104,26 @@ const CONFIG: BuildURLConfig = {
  */
 export function setBuildURLConfig(config: BuildURLConfig) {
   assert(`setBuildURLConfig: You must pass a config object`, config);
-  assert(`setBuildURLConfig: You must pass a config object with a 'host' or 'namespace' property`, 'host' in config || 'namespace' in config);
+  assert(
+    `setBuildURLConfig: You must pass a config object with a 'host' or 'namespace' property`,
+    'host' in config || 'namespace' in config
+  );
 
   CONFIG.host = config.host || '';
   CONFIG.namespace = config.namespace || '';
 
-  assert(`buildBaseURL: host must NOT end with '/', received '${CONFIG.host}'`, CONFIG.host === '/' || !CONFIG.host.endsWith('/'));
-  assert(`buildBaseURL: namespace must NOT start with '/', received '${CONFIG.namespace}'`, !CONFIG.namespace.startsWith('/'));
-  assert(`buildBaseURL: namespace must NOT end with '/', received '${CONFIG.namespace}'`, !CONFIG.namespace.endsWith('/'));
+  assert(
+    `buildBaseURL: host must NOT end with '/', received '${CONFIG.host}'`,
+    CONFIG.host === '/' || !CONFIG.host.endsWith('/')
+  );
+  assert(
+    `buildBaseURL: namespace must NOT start with '/', received '${CONFIG.namespace}'`,
+    !CONFIG.namespace.startsWith('/')
+  );
+  assert(
+    `buildBaseURL: namespace must NOT end with '/', received '${CONFIG.namespace}'`,
+    !CONFIG.namespace.endsWith('/')
+  );
 }
 
 export interface FindRecordUrlOptions {

